File: cases/taint/python/same-object-field-negative/same_object_field.py
class Holder:
    def __init__(self):
        self.tainted = "clean"
        self.clean = "clean"


def dfb_source():  # DFB-SOURCE: same-object-field-input
    return "tainted"


def dfb_sink(value):  # DFB-SINK: same-object-field-sink
    pass


def run():
    holder = Holder()
    holder.tainted = dfb_source()  # DFB-WITNESS: same-object-field-store
    holder.clean = "clean"
    dfb_sink(holder.clean)

File: cases/taint/ruby/same-object-field-positive/same_object_field.rb
class Holder
  attr_accessor :tainted, :clean

  def initialize
    @tainted = "clean"
    @clean = "clean"
  end
end

def dfb_source # DFB-SOURCE: same-object-field-input
  "tainted"
end

def dfb_sink(value) # DFB-SINK: same-object-field-sink
end

def run
  holder = Holder.new
  holder.tainted = dfb_source # DFB-WITNESS: same-object-field-store
  holder.clean = "clean"
  dfb_sink(holder.tainted)
end
